sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -344,6 +352,28 @@ object JavaTypeInference {
     }
   }
 
+  /** Returns a mapping from enum value to int for given enum type */
+  def enumSerializer[T](enum: Class[T]): T => Int = {
+    assert(enum.isEnum)
+    enum.getEnumConstants.toList.zipWithIndex.toMap
+  }
+
+  /** Returns value index for given enum type and value */
+  def serializeEnumName[T](enum: UTF8String, inputObject: T): Int = {
+    enumSerializer(Utils.classForName(enum.toString).asInstanceOf[Class[T]])(inputObject)
+  }
+
+  /** Returns a mapping from int to enum value for given enum type */
+  def enumDeserializer[T](enum: Class[T]): Int => T = {
```

Comment:
But it's still not very good, because we have to lookup enum by class name for each object we serialize/deserialize. Do you have any ideas?